pallets/creditcoin/src/lib.rs

```diff
@@ -608,10 +611,27 @@ pub mod pallet {
 			};
 
 			for (deadline, id, task) in PendingTasks::<T>::iter() {
-				let storage_key = (deadline, &id).encode();
-				let status = ocw::LocalVerificationStatus::new(&storage_key);
-				if status.is_complete() {
+				let storage_key = crate::ocw::tasks::storage_key(&id);
+				let offset =
+					T::UnverifiedTaskTimeout::get().saturated_into::<u32>().saturating_sub(2u32);
+
+				let mut lock = StorageLock::<BlockAndTime<frame_system::Pallet<T>>>::with_block_and_time_deadline(
+					&storage_key,
+					offset,
+					Duration::from_millis(0),
+				);
+
+				let guard = match lock.try_lock() {
+					Ok(g) => g,
+					Err(_) => continue,
+				};
+
+				if match &id {
+					TaskId::VerifyTransfer(id) => Transfers::<T>::contains_key(id),
+					TaskId::CollectCoins(id) => CollectedCoins::<T>::contains_key(id),
+				} {
 					log::debug!("Already handled Task ({:?}, {:?})", deadline, id);
+					guard.forget();
```

Comment:
We're missing test coverage for this line but I can add this later. The rest look good.